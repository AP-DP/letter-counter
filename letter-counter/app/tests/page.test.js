/**
 * @jest-environment jsdom
 */
import {expect, test} from "@jest/globals";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Home from "../page";

test("Intro renders correctly", () => {
  render(<Home />);
  const paragraph = screen.getByText((content, element) => content.startsWith('Get started'))
  expect(paragraph.innerHTML).toBe("Get started by typing into the box below.");
});

test('Page renders correctly', () => {
  const tree = renderer
    .create(<Home></Home>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});