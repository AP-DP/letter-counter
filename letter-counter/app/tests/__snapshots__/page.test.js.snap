// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<main
  className="main"
>
  <div
    className="description"
  >
    <p>
      Get started by typing into the box below.
    </p>
  </div>
  <div
    className="input"
    id="counterContainer"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div>
        <textarea
          className="form-control"
          disabled={false}
          onChange={[Function]}
          rows={10}
        />
        <div
          className="controls"
        >
          <button
            className="font-monospace btn btn-outline-primary"
            disabled={false}
            type="submit"
          >
            Submit
          </button>
          <button
            className="font-monospace btn btn-outline-primary"
            disabled={false}
            onClick={[Function]}
            type="reset"
          >
            Reset
          </button>
        </div>
      </div>
    </form>
    <div
      className="grid"
    >
      <div
        className="stack hstack"
        style={
          {
            "columnGap": "0px",
            "display": "grid",
            "gridTemplateColumns": "repeat(auto-fill, fit-content)",
          }
        }
      />
    </div>
  </div>
</main>
`;
